include/executors/executor_sycl.hpp

```diff
@@ -179,8 +178,8 @@ struct tree<using_shared_mem::disabled, tree_t, sharedMemT> {
   static void eval(tree_t &tree,
                    shared_mem<sharedMemT, using_shared_mem::disabled> scratch,
                    cl::sycl::nd_item<1> index) {
-    if ((index.get_global_id(0) < tree.getSize())) {  // FIXME:: this should move
-                                                   // to the tree not the root
+    if (tree.valid_thread(index)) {  // FIXME:: this should move
+                                     // to the tree not the root
```

Comment:
this valid thread is a bit confusing, the fixme should be a github issue.

Reply:
removed